File: app/src/main/java/com/example/dropletbarterapp/validators/Toaster.java
package com.example.dropletbarterapp.validators;

import android.content.Context;
import android.widget.Toast;

public class Toaster {

    public Boolean checkNullsAndGetToast(String message, Context context, String... checked) {
        for (String s : checked) {
            if (s == null) {
                Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
                return false;
            }
        }
        return true;
    }

    public void getToast(Context context, String message){
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
    }

    public Boolean checkNullsAndGetToast(String message, Context context, String checked) {
        if (checked == null) {
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
            return false;
        }
        return true;
    }

}

File: app/src/main/java/com/example/dropletbarterapp/validators/Toaster.kt
package com.example.dropletbarterapp.validators

import android.content.Context
import android.widget.Toast

class Toaster() : ErrorSender {

    override val validator: Validator
        get() = Validator()

    fun getToast(context: Context, message: String) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }

    override fun checkNullsAndGetMessage(
        context: Context,
        message: String,
        field: String?
    ): Boolean {
        return if (validator.validateFieldIsNull(field)) {
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
            false
        } else {
            true
        }

    }

    override fun checkNullsAndGetMessage(
        context: Context,
        message: String,
        vararg field: String?
    ): Boolean {
        for (f in field) {
            if (validator.validateFieldIsNull(f)) {
                Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
                return false
            }
        }
        return true
    }

    override fun showMessage(context: Context, message: String) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }
}